routing/road_graph_router.cpp

```diff
@@ -212,16 +212,17 @@ IRouter::ResultCode RoadGraphRouter::CalculateRoute(m2::PointD const & startPoin
   m_roadGraph->AddFakeEdges(startPos, startVicinity);
   m_roadGraph->AddFakeEdges(finalPos, finalVicinity);
 
-  vector<Junction> path;
+  RoutingResult<Junction> result;
   IRoutingAlgorithm::Result const resultCode =
-      m_algorithm->CalculateRoute(*m_roadGraph, startPos, finalPos, delegate, path);
+      m_algorithm->CalculateRoute(*m_roadGraph, startPos, finalPos, delegate, result);
 
   if (resultCode == IRoutingAlgorithm::Result::OK)
   {
-    ASSERT(!path.empty(), ());
-    ASSERT_EQUAL(path.front(), startPos, ());
-    ASSERT_EQUAL(path.back(), finalPos, ());
-    ReconstructRoute(move(path), route, delegate);
+    ASSERT(!result.path.empty(), ());
+    ASSERT_EQUAL(result.path.front(), startPos, ());
+    ASSERT_EQUAL(result.path.back(), finalPos, ());
+    ASSERT_GREATER(result.distance, 0., ());
```

Comment:
OK here, but I general we should not restrict our algorithms in such way, because by some circumstances somebody may ask router to build a route from some point to itself, and the router should not crash, because empty pass is a valid path.